jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java

```diff
@@ -1953,4 +1987,56 @@ public X509 getX509()
             return _x509;
         }
     }
+
+    public static class Client extends SslContextFactory
+    {
+        public Client()
+        {
+            this(false);
+        }
+
+        public Client(boolean trustAll)
+        {
+            super(trustAll);
+        }
+
+        @Override
+        protected void checkConfiguration()
+        {
+            checkTrustAll();
+            checkEndPointIdentificationAlgorithm();
```

Comment:
Should we warn if someone sets up wantClientAuth or needClientAuth to true in the Client mode?

Reply:
Don't know, it's not really an "insecure" configuration, just one that has no effect.

Eventually in Jetty 10 `SslContextFactory` will be `abstract` and those methods only present in `Server`.

Reply:
Good point, do we have test cases still around that keep the old `new SslContextFactory()` usage?

Reply:
No, now we do not have code that uses `SslContextFactory` directly, it's all either `Client` or `Server`.